src/main/java/io/confluent/idesidecar/restapi/clients/SchemaRegistryClients.java

```diff
@@ -85,12 +86,21 @@ private SchemaRegistryClient createClient(
       Map<String, Object> configurationProperties,
       Map<String, String> headers
   ) {
+    var restService = new RestService(srClusterUri);
+    restService.configure(configurationProperties);
+    restService.setHttpHeaders(headers);
+
+    var sslFactory = new SslFactory(configurationProperties);
+    if (sslFactory.sslContext() != null) {
+      restService.setSslSocketFactory(sslFactory.sslContext().getSocketFactory());
+    }
+
```

Comment:
If we don't instantiate the RestService this way and pass it into the `CachedSchemaRegistryClient`, it will fail with `unable to find valid certification path to requested target`. I have no idea why. 

How did I land at this solution? I'd extracted out the code from the constructor to debug what was being set in the `sslFactory` and before I knew it, that had fixed the issue. 🤷 I'll take it.